### PR TITLE
(#156)New job opportunity should get created on ColoredCow website also - feature

### PR DESCRIPTION
**Pull request by atul5238**
Targets #156 
## Summary

1. New feature integrated, which will Create/Update/Delete the job opportunities on the ColoredCow website also.
2. Feature is created using Corcel.

### CODE REVIEW

File: config/corcel.php [hunk 1]
```diff
@@ -31,7 +45,7 @@
     */
 
     'post_types' => [
-//        'video' => App\Models\Video::class,
+    //    'career' => App\Models\Career::class,
```
[the remark this answers is not in the document]
Comment (atul5238) [reply config/corcel.php:48]: No, I changed this at the time of testing in my local but commented if afterwards

File: config/database.php [hunk 2]
```diff
@@ -92,15 +92,29 @@
...
             'port' => env('DB_PORT', '1433'),
-            'database' => env('DB_WORDPRESS_DATABASE', ''),
-            'username' => env('DB_WORDPRESS_USERNAME', 'forge'),
+            'database' => env('DB_WORDPRESS_DATABASE', 'coloredcow_db'),
+            'username' => env('DB_WORDPRESS_USERNAME', 'root'),
             'password' => env('DB_WORDPRESS_PASSWORD', ''),
             'charset' => 'utf8',
-            'collation' => 'utf8_unicode_ci',
-            'prefix' => env('DB_WORDPRESS_PRIFIX', 'wp_'),
+            'collation' => 'utf8_general_ci',
```
[the remark this answers is not in the document]
Comment (atul5238) [reply config/database.php:99]: utf8_general_ci is database type in my local-database for Website

File: config/database.php [hunk 3]
```diff
@@ -92,15 +92,29 @@
...
+            'prefix' => env('DB_WORDPRESS_PRIFIX', 'cc_'),
             'strict' => false,
             'engine' => null,
         ],
+
+        'corcel' => [ // for WordPress database (used by Corcel)
```
[the remark this answers is not in the document]
Comment (atul5238) [reply config/database.php:105]: This is duplicate code, left at the time of testing. Removing this.

File: config/app.php [hunk 4]
```diff
@@ -162,6 +162,8 @@
         Spatie\Permission\PermissionServiceProvider::class,
         App\Providers\DatabaseServiceProvider::class,
         OfficeSuite\OfficeSuitesServiceProvider::class,
+        Corcel\Laravel\CorcelServiceProvider::class,
+
```
Comment (rathorevaibhav) [inline config/app.php:166]: remove extra line

File: config/corcel.php [hunk 5]
```diff
@@ -14,6 +14,20 @@
...
     |
     */
+    
+    'corcel' => [ // for WordPress database (used by Corcel)
+        'driver'    => 'mysql',
+        'host'      => env('DB_HOST', '127.0.0.1'),
+        'port'      => env('DB_PORT', '3306'),
+        'database'  => env('DB_WP_DATABASE', 'coloredcow_db'),
+        'username'  => env('DB_USERNAME', 'root'),
+        'password'  => env('DB_PASSWORD', ''),
```
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply config/corcel.php:24]: I don't see this has been fixed yet @atul5238

File: config/database.php [hunk 6]
```diff
@@ -92,15 +92,29 @@
             'driver' => 'mysql',
             'host' => env('DB_HOST', 'localhost'),
             'port' => env('DB_PORT', '1433'),
-            'database' => env('DB_WORDPRESS_DATABASE', ''),
-            'username' => env('DB_WORDPRESS_USERNAME', 'forge'),
+            'database' => env('DB_WORDPRESS_DATABASE', 'coloredcow_db'),
```
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply config/database.php:95]: I don't see this has been addressed yet @atul5238

File: config/database.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply config/database.php:99]: Are you sure it will work on production/uat? They are Ubuntu boxes

**Review by rathorevaibhav — COMMENTED** [review]
@atul5238 I still see some of the feedbacks haven't been addressed yet. Please check.

File: config/database.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (atul5238) [reply config/database.php:99]: I tested it using utf8_unicode_ci and it is working fine. So for production we can keep this default i.e 'utf8_unicode_ci'.

File: config/database.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (atul5238) [reply config/database.php:95]: Here we have to replace forge with my localhost database username for WP. I tried with forge but it said this username doesn't exist.

File: config/app.php (hunk 4 above)
Comment (atul5238) [reply config/app.php:165]: removed in the latest commit.

File: config/database.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply config/database.php:95]: This change is not needed
You need to define the `DB_WORDPRESS_DATABASE` in your `.env` file.

```
DB_WORDPRESS_DATABASE=coloredcow_db
```

File: config/corcel.php [hunk 7]
```diff
@@ -14,6 +14,20 @@
...
+        'port'      => env('DB_WP_PORT', '3306'),
+        'database'  => env('DB_WP_DATABASE', 'coloredcow_db'),
+        'username'  => env('DB_WP_USERNAME', 'root'),
+        'password'  => env('DB_WP_PASSWORD', ''),
+        'charset'   => 'utf8mb4',
+        'collation' => 'utf8mb4_unicode_ci',
+        'prefix'    => 'wp_',
+        'strict'    => false,
+        'engine'    => null,
+    ],
```
Comment (rathorevaibhav) [inline config/corcel.php:30]: I don't think this is needed. You have already defined this in database.php

How is it working differently?

File: config/database.php [hunk 8]
```diff
@@ -92,15 +92,16 @@
...
+            'username' => env('DB_WORDPRESS_USERNAME', 'root'),
             'password' => env('DB_WORDPRESS_PASSWORD', ''),
             'charset' => 'utf8',
             'collation' => 'utf8_unicode_ci',
             'prefix' => env('DB_WORDPRESS_PRIFIX', 'wp_'),
             'strict' => false,
             'engine' => null,
         ],
+
```
Comment (rathorevaibhav) [inline config/database.php:104]: remove unnecessary changes.

**Review by rathorevaibhav — COMMENTED** [review]
@atul5238 final feedbacks

Also, there is a submodule in your Pull Request (Modules/HR). It is not needed. Please remove it also.

File: app/Http/Controllers/CareerController.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Career;
+use Illuminate\Http\Request;
+
+class CareerController extends Controller
```
[the remark this answers is not in the document]
Comment (atul5238) [reply app/Http/Controllers/CareerController.php]: Created this at the time first approach. Removing it now

File: config/corcel.php (hunk 7 above)
Comment (atul5238) [reply config/corcel.php:30]: this is duplicate, removing this also

File: config/database.php (hunk 8 above)
Comment (atul5238) [reply config/database.php:104]: removing

Comment (rathorevaibhav) [reply config/database.php:104]: @atul5238 I still see this hasn't been removed.

File: config/database.php [hunk 10]
```diff
@@ -92,15 +92,29 @@
...
-            'database' => env('DB_WORDPRESS_DATABASE', ''),
-            'username' => env('DB_WORDPRESS_USERNAME', 'forge'),
+            'database' => env('DB_WORDPRESS_DATABASE', 'coloredcow_db'),
+            'username' => env('DB_WORDPRESS_USERNAME', 'root'),
             'password' => env('DB_WORDPRESS_PASSWORD', ''),
             'charset' => 'utf8',
-            'collation' => 'utf8_unicode_ci',
-            'prefix' => env('DB_WORDPRESS_PRIFIX', 'wp_'),
+            'collation' => 'utf8_general_ci',
+            'prefix' => env('DB_WORDPRESS_PRIFIX', 'cc_'),
```
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply config/database.php:100]: @atul5238 I still see this hasn't been updated.

**Review by rathorevaibhav — COMMENTED** [review]
@atul5238 please check all the feedbacks.
